linux/LinuxProcessList.c

```diff
@@ -1824,8 +1824,33 @@ static inline double LinuxProcessList_scanCPUTime(ProcessList* super) {
    return period;
 }
 
+static bool cpuIsOnline(unsigned int cpuId) {
+      /* if there's no sysfs file for the CPU node indicating online/offline status,
+       * the CPU is probably online, so assume online by default
+       */
+      bool online = 1;
+      char pathBuffer[64];
+
+      xSnprintf(pathBuffer, sizeof(pathBuffer), "/sys/devices/system/cpu/cpu%u/online", cpuId);
+
+      FILE* file = fopen(pathBuffer, "r");
```

Comment:
Please use the wrapper function for `openat` or `readfile` to reduce the number of FS level operations.